dvc/cache/base.py

```diff
@@ -101,23 +120,33 @@ def changed(self, path_info, hash_info):
         Args:
             path_info: dict with path information.
             hash: expected hash value for this data.
+            filter_info: an optional argument to target a specific path.
 
         Returns:
             bool: True if data has changed, False otherwise.
         """
 
+        path = filter_info or path_info
         logger.trace(
             "checking if '%s'('%s') has changed.", path_info, hash_info
         )
 
-        if not self.tree.exists(path_info):
-            logger.debug("'%s' doesn't exist.", path_info)
+        if not self.tree.exists(path):
+            logger.debug("'%s' doesn't exist.", path)
             return True
 
         if not hash_info:
             logger.debug("hash value for '%s' is missing.", path_info)
             return True
 
+        if path != path_info:
+            filtered_hash_info = self._get_filtered_hash_info(
+                path, hash_info, path_info
+            )
+            if filtered_hash_info is not None:
+                path_info = filter_info
```

Comment:
Might be missing something, but how about we don't do this assignment and just use `path` down below to avoid confusion? 

We'll need to do something like `if filtered_hash_info is None: path = path_info` as a fallback though.

Reply:
@isidentical `path_info` is coming in as an argument, so feels a bit weird to have `path` and then also overwrite `path_info`, which results in a confusion of not having to figure out if `path_info` is what came to us or is the new filtered value.

Reply:
```
        if path != path_info:
            filtered_hash_info = self._get_filtered_hash_info(
                path, hash_info, path_info
            )
            if filtered_hash_info is None:
                path = path_info
            else:
                hash_info = filtered_hash_info
```
something like this?